cmd/release-controller/sync_analysis.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	imagev1 "github.com/openshift/api/image/v1"
+	"k8s.io/klog"
+)
+
+func (c *Controller) ensureAnalysisJobs(release *Release, releaseTag *imagev1.TagReference) error {
+	for name, analysisType := range release.Config.Analysis {
+		if analysisType.Disabled {
+			klog.V(2).Infof("Release analysis step %s is disabled, ignoring", name)
+			continue
+		}
+		jobCount := 10
+		if analysisType.AnalysisJobCount > 0 && analysisType.AnalysisJobCount <= 20 {
+			jobCount = analysisType.AnalysisJobCount
+		}
+
+		// TODO: How do we want to track these jobs...
+		jobLabels := map[string]string{
+			"release.openshift.io/analysis": releaseTag.Name,
+		}
+
+		switch {
+		case analysisType.ProwJob != nil:
+			// if this is an upgrade job, find the appropriate source for the upgrade job
+			var previousTag, previousReleasePullSpec string
+			if analysisType.Upgrade {
```

Comment:
as a newcomer to this repo, structure like this surprises me.  Are there refactoring PRs in progress to make this more idiot proof?

Reply:
We agreed on a followup